src/neuroconv/datainterfaces/text/timeintervalsinterface.py

```diff
@@ -0,0 +1,133 @@
+from abc import abstractmethod
+import os
+
+from typing import Dict, Optional
+
+import numpy as np
+import pandas as pd
+from pynwb import NWBFile
+from pynwb.epoch import TimeIntervals
+
+from ...basedatainterface import BaseDataInterface
+from ...utils.types import FilePathType
+from ...tools.nwb_helpers import make_or_load_nwbfile
+from ...utils.dict import load_dict_from_file
+
+
+def convert_df_to_time_intervals(
+    df: pd.DataFrame,
+    name: str = "trials",
+    description: Optional[str] = None,
+    column_name_mapping: Dict[str, str] = None,
+    column_descriptions: Dict[str, str] = None,
+):
+    """
+    Convert a dataframe to a TimeIntervals object.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        The dataframe to convert.
+    name : str, optional
+        The name of the TimeIntervals object.
+    description : str, optional
+        The description of the TimeIntervals object.
+    column_name_mapping: dict, optional
+        If passed, rename subset of columns from key to value.
+    column_descriptions: dict, optional
+        Keys are the names of the columns (after renaming) and values are the descriptions. If not passed,
+        the names of the columns are used as descriptions.
+
+    Returns
+    -------
+    TimeIntervals
+
+    """
```

Comment:
Our usual organization would include the import for this function under `neuroconv.tools.text` in case someone wanted to use it as a handy helper independently of the data interfaces or other NeuroConv API

Reply:
ok, moved.